Tools/Scripts/webkitpy/layout_tests/controllers/test_result_writer.py

```diff
@@ -100,18 +100,22 @@ def output_filename(self, modifier):
 
         ext_parts = fs.splitext(self._test_name)
         output_basename = ext_parts[0]
+        extension = ext_parts[1]
+
+        for char in ('?', '#'):
+            index = extension.find(char)
+            if index != -1:
+                extension = extension[:index]
```

Comment:
Where is `extension` used?

Reply:
Hum, good question, I wonder if it was intended to be used instead of `ext_parts[1]` below. I'll take a look!

Reply:
That's can't be the case, though, because otherwise we'd end up clobbering the same file name for different test variants. It looks like the intent of this code is to change something like `foo/bar.html?variant=1` to have a result file of something like `foo/bar_variant=1-expected.txt`

Reply:
Yes, I updated the patch and added a comment about this, hopefully it looks better now.